src/object_store/src/object/opendal_engine/opendal_s3.rs

```diff
@@ -78,4 +79,56 @@ impl OpendalObjectStore {
 
         Ok(HttpClient::build(client_builder)?)
     }
+
+    /// currently used by snowflake sink,
+    /// especially when sinking to the intermediate s3 bucket.
+    pub fn new_s3_engine_with_credentials(
+        bucket: &str,
+        object_store_config: ObjectStoreConfig,
+        aws_access_key_id: &str,
+        aws_secret_access_key: &str,
+        aws_region: &str,
+    ) -> ObjectResult<Self> {
+        // Create s3 builder with credentials.
+        let mut builder = S3::default();
+
+        // set credentials for s3 sink
+        builder.bucket(bucket);
+        builder.access_key_id(aws_access_key_id);
+        builder.secret_access_key(aws_secret_access_key);
+        builder.region(aws_region);
+
+        // For AWS S3, there is no need to set an endpoint; for other S3 compatible object stores, it is necessary to set this field.
+        if let Ok(endpoint_url) = std::env::var("RW_S3_ENDPOINT") {
+            builder.endpoint(&endpoint_url);
+        }
+
+        if std::env::var("RW_IS_FORCE_PATH_STYLE").is_err() {
+            builder.enable_virtual_host_style();
+        }
+
+        let http_client = Self::new_http_client(&object_store_config)?;
+        builder.http_client(http_client);
+
+        let op: Operator = Operator::new(builder)?
+            .layer(LoggingLayer::default())
+            .layer(
+                RetryLayer::new()
+                    .with_min_delay(Duration::from_millis(
+                        object_store_config.s3.object_store_req_retry_interval_ms,
+                    ))
+                    .with_max_delay(Duration::from_millis(
+                        object_store_config.s3.object_store_req_retry_max_delay_ms,
+                    ))
+                    .with_max_times(object_store_config.s3.object_store_req_retry_max_attempts)
+                    .with_factor(1.1)
```

Comment:
the original factor is `1.0`, but strange error will be thrown for `RetryLayer`, e.g., saying that factor can't be lower then `1.0` - thus I changed it to `1.1`.